### PR TITLE
fix: remove gene following call for logged out experience

### PR DESCRIPTION
**Pull request by egdbear**
The type of this PR is: **Fix**

This PR solves [DIA-845]

### Description

Screenshot 2024-09-17 at 10 41 49 AM

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DIA-845]: https://artsyproduct.atlassian.net/browse/DIA-845?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by araujobarret — COMMENTED** [review]
I think if the purpose of it is just to get rid of the extra request instead of scrapping the data from the page to use with `FollowGeneButton` we can just use the fragment directly since it depends on gene data already and then remove this query renderer element.

**Comment by egdbear** [comment]
> I think if the purpose of it is just to get rid of the extra request instead of scrapping the data from the page to use with `FollowGeneButton` we can just use the fragment directly since it depends on gene data already and then remove this query renderer element.

Yeah that sounds like a plan!

**Review by araujobarret — APPROVED** [review]

File: src/Apps/Gene/Routes/GeneShow.tsx [hunk 1]
```diff
@@ -26,7 +26,7 @@ export const GeneShow: React.FC<GeneShowProps> = ({ gene }) => {
             {gene.displayName || gene.name}
           </Text>
 
-          <FollowGeneButtonQueryRenderer id={gene.internalID} />
+          <FollowGeneButtonFragmentContainer gene={gene} />
```
Comment (mzikherman) [inline src/Apps/Gene/Routes/GeneShow.tsx:29]: I'm not sure if this is the right change? You switched away from deferring those calls to requiring that data up-front in order to initially render the page (adding `isFollowed` to the fragment).

Would it be better to keep this as a deferred renderer (that's what we want to do for personalized content, generally) - and simply return early from there if there's no user?

**Comment by damassi** [comment]
@egdbear - can we be sure to write proper PR descriptions? Linking out to JIRA and providing titles alone doesn't supply enough context within the github / repo ecosystem for future devs to understand the change.

**Comment by egdbear** [comment]
> @egdbear - can we be sure to write proper PR descriptions? Linking out to JIRA and providing titles alone doesn't supply enough context within the github / repo ecosystem for future devs to understand the change.

of course, sorry this was already discussed on the slack thread, will add the description!

File: src/Apps/Gene/Routes/GeneShow.tsx (hunk 1 above)
Comment (egdbear) [reply src/Apps/Gene/Routes/GeneShow.tsx:29]: I think that sounds more appealing! (and somehow similar to the first implementation)

Comment (araujobarret) [reply src/Apps/Gene/Routes/GeneShow.tsx:29]: and most of the data(besides isFollowed) is already fetched by the components calling the former `FollowGeneButtonQueryRenderer`.
We talked about having the `isFollowed` on MP as a fallback to always `false` if the user is not logged in to save from hitting the loader when not needed too

Comment (mzikherman) [reply src/Apps/Gene/Routes/GeneShow.tsx:29]: We want personalized data (like `isFollowed`) to be in its own query renderer. This allows those queries to skip the cache, and any other page-data queries (like gene page, a list of genes, etc.) to be cacheable (even for logged-in).

So keeping follow buttons in their own query renderer _is_ what we want.

Seems like it should be not hard to suppress the call it was making (and MP immediately responds with `isFollowed: false`), but I haven't really checked out the code!

If there's some complexity where an early return _skips rendering the follow button altogether_ - then yea, maybe there needs to be a wrapper that renders _either_ the query renderer component _or_ the 'default' follow button in the un-saved state?

(something like that - I'm being a touch hand-wave-y)

Comment (araujobarret) [reply src/Apps/Gene/Routes/GeneShow.tsx:29]: I think I'm with you but all the data we query before(slug, internalID) should come from the fragment, now the personalized data totally agree with the separate query.
How about a compromise, we keep the other un-personalized data in the fragment and add a lazyLoad in the same element just to query the personalized data(isFollowed)?

re MP we took a quick look the the field `isFollowed` his a loader there, must be simple to fallback false given unauthenticated users

Comment (mzikherman) [reply src/Apps/Gene/Routes/GeneShow.tsx:29]: Yep @araujobarret that sounds good! I'm glossing over the slug/id/name part as well - that's of course totally fine to include up-front.

**Comment by damassi** [comment]
Thanks, yah, I try to think about PRs as artifacts best viewed 3-5 years in the future. At that point nobody will know what slack convo something referenced, and JIRA might be gone, and so on and so forth. So Github should remain canonical as the source of truth 🙏

**Pull request merged**